pkg/controller/virtualmachineimport/virtualmachineimport_controller.go

```diff
@@ -139,6 +139,26 @@ func (r *ReconcileVirtualMachineImport) Reconcile(request reconcile.Request) (re
 		return reconcile.Result{}, err
 	}
 
+	// Load the external resource mapping
+	resourceMapping, err := r.fetchResourceMapping(&instance.Spec.ResourceMapping)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Prepare/merge the resourceMapping
+	err = provider.PrepareResourceMapping(resourceMapping, instance.Spec.Source)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Validate if it's needed at this stage of processing
+	if shouldValidate(&instance.Status) {
+		err = provider.Validate()
```

Comment:
I intended to update the status within this context instead on providers, so we'll have a unified  behavior to all providers.
I'm okay to accept this as a template and re-evaluate later on when we'll start updating the status